### ISSUE TITLE
Incorrect GB pointer calculation

### ISSUE THREAD

**Issue by HoratioVex** [stevemonaco/Atlas#1]
It seems the calculation of the Gameboy memory mapping via the "GB" pointer type uses a wrong formula.

Pointer::GetGBAddress uses:

   Bank = Offset / 0x4000;
   Word = Offset % ((Bank+1) * 0x4000);

which only returns itself and is only accidentally correct for banks 0 and 1.

Correct would be (for banks >0):
   Word=Offset+0x4000-(Bank*0x4000)

**Comment by stevemonaco** [comment]
Hi. Yeah, this looks to be a bug. I was trying to avoid conditional logic 20 years ago and screwed up. AFAIK, this is the only report.

If you can't clone+compile Atlas to make the changes locally, I would suggest trying abcde. You will need to modify the source as the same bug seems to have been ported there in Atlas.pm:

```perl
} elsif ($addressType eq 'GB') {
	my $bank = int($value / 0x4000) & 0xFFFF;
	my $word = ($value % (($bank + 1) * 0x4000)) & 0xFFFF;
	$value = $bank * 0x10000 + $word;
```

I don't work on Atlas anymore so I'll leave this issue up. I'd accept a PR for a fix, but probably won't make a new release.

**Comment by HoratioVex** [comment]
Ok, I sent a PR. It also includes the script to activate automatic Github builds, because they're great and I don't know how to filter out single commits. 😄 

The main issue will be RHDN though. I suspect they'll remain the main download source, and they'll never accept an update from someone who isn't the official author. So lots of future romhackers will bang their head against this bug...

Also I sent abw a note if he wants to fix his perl script.

**Comment by stevemonaco** [comment]
Closed by #2 

I will try to make some updates here and maybe get a new version on RHDN. The VC++ runtimes should probably be updated and statically linked.
